packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -102,30 +104,14 @@ _Props/Attrs_
 
 -   `field-name`: string
 -   `key-type`: 'string' | 'number' | 'boolean'
+-   `display`: enum with possible values of `iconOnly`, `labelOnly`, and `iconAndLabel` - Specifies what is rendered in a cell for the column. Note, that this attribute only affects a cell's rendering; a group row will always render both the icon and the label.
+-   `icon-width`: boolean - When `true`, the column will have a fixed width that makes the column the appropriate width to render only a single icon in the cell. This should only be set to `true` when the header contains a single icon (no text) and `display` is `iconOnly`. If set to `true` when `display` is not `iconOnly`, the column's configuration will be considered invalid.
```

Comment:
We tend to not do "magical" things like that. We also don't do anything today with the content slotted in a column header. Therefore, I'm leaning towards leaving this as a property on the column so that clients can explicitly opt-in to this behavior. However, I'll leave the conversation open for more discussion.